ripley/lib/canvas_utils.py

```diff
@@ -578,12 +585,19 @@ def _add_instructor_to_site(
 
     user_section_feed = explicit_sections_for_instructor[0] if explicit_sections_for_instructor else section_feeds[0]
     sis_section_id = user_section_feed['section_id']
-    canvas_section = canvas.get_section(section_id=f'sis_section_id:{sis_section_id}', api_call=False, use_sis_id=True)
+    canvas_section = canvas.get_section(
+        api_call=False,
+        section_id=f'sis_section_id:{sis_section_id}',
+        use_sis_id=True,
+    )
+    instructor_role_id = section_roles[section_feeds[0]['section_id']] or (teacher_role and teacher_role.id)
     canvas_section.enroll_user(
-        sis_user_profile['id'],
-        role_id=instructor_role_id,
-        enrollment_state='active',
-        notify=False,
+        user=sis_user_profile['id'],
+        ** {
+            'enrollment[role_id]': instructor_role_id,
+            'enrollment[enrollment_state]': 'active',
+            'enrollment[notify]': False,
+        },
```

Comment:
This is the fix. The stuff above is improved code-readability. Below is better UI-ish.